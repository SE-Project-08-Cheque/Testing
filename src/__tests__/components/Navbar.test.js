// import dependencies
import React from 'react';
// import react dom
import ReactDOM from 'react-dom';
// import react-testing methods
import {render, cleanup, fireEvent} from '@testing-library/react';
// add custom jest matchers from jest-dom
import "@testing-library/jest-dom/extend-expect";
// import renderer for take snapshots
import renderer from 'react-test-renderer';

// import the component for testing
import Navbar from '../../components/Navbar';



afterEach(cleanup);

it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Navbar heading="onlinecheque"/>, div)
})

it("renders navbar buttons correctly", () => {
    const {getByTestId} = render(<Navbar />)
    expect(getByTestId('home')).toBeTruthy();
    expect(getByTestId('about_us')).toBeInTheDocument();
    expect(getByTestId('log_out')).toBeTruthy();

    
})

it("buttons links correct", () => {
    const {getByText} = render(<Navbar />)
    expect(getByText("Home").href).toBe("http://localhost/")
    expect(getByText("About Us").href).toBe("https://drive.google.com/file/d/1AdNMc9rVaYs0j5dFtpDZk72LFPk_h6Hv/view?usp=sharing")
})
describe("ClickButton", () => {
        it("onClick", () => {
            const {queryByTestId} = render(<Navbar />)
            expect(queryByTestId("log_out")).toHaveTextContent("Log out"); 
            const btn = queryByTestId("log_out");
    fireEvent.click(btn)
            // expect(queryByTestId("asd").innerHTML).toHaveTextContent("yOU Clicked"); 
        })
    })

// it('routes to a new route', async () => {
//     const history = createMemoryHistory();
  
//     // mock push function
//     history.push = jest.fn();
  
//     const { getByText } = render(
//       <MemoryRouter history={history}>
//         <Link to="/">Home</Link>
//       </MemoryRouter>
//     );
  
//     fireEvent.click(getByText('Home'));
  
//     // spy on push calls, assert on url (parameter)
//     expect(history.push).toHaveBeenCalledWith("/");
//   });

it("matches snapshot", () => {
    const tree = renderer.create(<Navbar />).toJSON();
    expect(tree).toMatchSnapshot();
})