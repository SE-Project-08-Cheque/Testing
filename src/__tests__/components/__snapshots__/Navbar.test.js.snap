// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<nav
  className="css-pex5or"
>
  <div
    className="css-1c714fi"
  >
    <a
      href="/"
    >
      <h1
        className="chakra-heading css-1nmn3av"
      >
        <i
          className="fab fa-typo3"
        />
      </h1>
    </a>
  </div>
  <div
    className="css-dsyim5"
    onClick={[Function]}
  >
    <svg
      className="chakra-icon css-onkibi"
      focusable={false}
      viewBox="0 0 24 24"
    >
      <path
        d="M 3 5 A 1.0001 1.0001 0 1 0 3 7 L 21 7 A 1.0001 1.0001 0 1 0 21 5 L 3 5 z M 3 11 A 1.0001 1.0001 0 1 0 3 13 L 21 13 A 1.0001 1.0001 0 1 0 21 11 L 3 11 z M 3 17 A 1.0001 1.0001 0 1 0 3 19 L 21 19 A 1.0001 1.0001 0 1 0 21 17 L 3 17 z"
        fill="currentColor"
      />
    </svg>
  </div>
  <div
    className="css-1f3o17h"
  >
    <button
      className="chakra-button css-1ohh2gv"
      type="button"
    >
      <a
        data-testid="home"
        href="/"
      >
        Home
      </a>
    </button>
    <button
      className="chakra-button css-1ohh2gv"
      type="button"
    >
      <a
        data-testid="about_us"
        href="https://drive.google.com/file/d/1AdNMc9rVaYs0j5dFtpDZk72LFPk_h6Hv/view?usp=sharing"
        target="_blank"
      >
        About Us
      </a>
    </button>
  </div>
  <div
    className="css-iqqio3"
    data-testid="navbar"
  >
    <button
      aria-label="Call Segun"
      className="chakra-button css-w6lnh7"
      type="button"
    >
      <svg
        aria-hidden={true}
        className="chakra-icon css-onkibi"
        focusable={false}
        viewBox="0 0 14 14"
      >
        <path
          d="M14,7.77 L14,6.17 L12.06,5.53 L11.61,4.44 L12.49,2.6 L11.36,1.47 L9.55,2.38 L8.46,1.93 L7.77,0.01 L6.17,0.01 L5.54,1.95 L4.43,2.4 L2.59,1.52 L1.46,2.65 L2.37,4.46 L1.92,5.55 L0,6.23 L0,7.82 L1.94,8.46 L2.39,9.55 L1.51,11.39 L2.64,12.52 L4.45,11.61 L5.54,12.06 L6.23,13.98 L7.82,13.98 L8.45,12.04 L9.56,11.59 L11.4,12.47 L12.53,11.34 L11.61,9.53 L12.08,8.44 L14,7.75 L14,7.77 Z M7,10 C5.34,10 4,8.66 4,7 C4,5.34 5.34,4 7,4 C8.66,4 10,5.34 10,7 C10,8.66 8.66,10 7,10 Z"
          fill="currentColor"
        />
      </svg>
    </button>
    <button
      className="chakra-button css-apt07k"
      data-testid="log_out"
      onClick={[Function]}
      type="button"
    >
      Log out
    </button>
  </div>
</nav>
`;
